src/server/pa_rot.rs

```diff
@@ -217,8 +241,73 @@ where
                 })
                 .map_err(|_| UNSPECIFIED)?;
 
+                if let Some(KeyExchangeAlgo::Ecdh) = ctx.server.key_exchange {
+                    ctx.server
+                        .opts
+                        .session
+                        .create_session(ctx.req.nonce, tbs.nonce)
+                        .map_err(|_| UNSPECIFIED)?;
+                    ctx.server.current_cert_slot = Some(tbs.slot);
+                }
+
                 Ok(ChallengeResponse { tbs, signature })
             })
+            .handle::<protocol::KeyExchange, _>(|ctx| {
```

Comment:
I've been trying to think of a good strategy for this, because it's a huge pain to edit too. Would you be ok approving this if I said that I plan to deal with this, for all the handlers in this file, with a followup?

Reply:
Yes, that's fine with me.